Apps/Src/ReadCarCAN.c

```diff
@@ -30,22 +31,24 @@ void Task_ReadCarCAN(void *p_arg) {
 
         // Normal task countinues here
 
+        uint32_t canId;
+
         // Check if BPS sent us a message
-        if (CANbus_Read(buffer) == SUCCESS) {
-            long msg = *((long *)(&buffer[0]));
-            if (msg == 0) {
-                car->IsRegenBrakingAllowed = OFF;
+        if (CANbus_Read(&canId, buffer, CAN_NON_BLOCKING) == SUCCESS) {
+            // If charge_enable, set regen flag
+            if (canId == CHARGE_ENABLE) {
+                car->IsRegenBrakingAllowed = (buffer[0] == 0) ? OFF : ON;
+                Contactors_Set(ARRAY, car->IsRegenBrakingAllowed);
+                faultCounter = 0;
             } else {
-                car->IsRegenBrakingAllowed = ON;
+                // If we didn't get a message, something might have gone wrong
+                faultCounter++;
```

Comment:
Doesn't this line only get run if we receive an CAN message with an id other that `CHARGE_ENABLE`? This code will not run if you aren't receiving any CAN messages